### ISSUE TITLE
Screens immediately waking up after going to sleep when app is running

### ISSUE THREAD

**Issue by evelyndooley** [kangyu-california/PersistentWindows#98]
Issue: When the my screens turn off due to inactivity, they only stay off for maybe 10 seconds before waking themselves up again. This only happens when PersistentWindows is running. 

Version: 5.18.7

I just got a new monitor and every time the screens go to sleep, when I move the mouse to wake it up again all my window positions get moved to the primary screen. This didn't happen before, but I could not find a way around it so I installed this app. Now, the window positions are properly positioned when the screens wake, however they wake up right away as described.

**Comment by kangyu-california** [comment]
First clarification is PW does not wake up a blank screen, it is Windows OS that does the job.

Secondly I admit the screen wake-up might be a side-effect of PW responding to display resolution change by initiating auto-restore.

You may confirm this by searching for event 9990 and 9999 during that critical 10 second timeframe in event viewer, for example
![image](https://user-images.githubusercontent.com/59128756/111351356-55c5fe00-8640-11eb-980e-623c8ed236d6.png)

The easiest way to avoid PW auto-restore is do NOT wakeup a screen sleep using mouse click,  while the PC is still awake, even mouse move might trigger some window event, causing PW to enable auto-restore for the blank screen configuration, which is undesirable.

Based on that assumption, the first workaround is to restart PW, and consistently wake up screen sleep using some key that is unlikely to trigger window event, I would suggest using CAPs key

If the above workaround is still not working, please try to "Pause auto restore" from PW menu before leaving PC idle.

**Comment by evelyndooley** [comment]
When this happens these messages are generated:
```
PersistentWindows: No need to restore Display_Loc0x0_Res2560x1440 display session
PersistentWindows: Start restoring window layout back to 3/16/2021 5:06:36 PM for display setting Display_Loc0x0_Res2560x1440__Display_Loc2560xM345_Res2560x1440
PersistentWindows: Restore halted due to new display setting change Display_Loc0x0_Res2560x1440
PersistentWindows: Start restoring window layout back to 3/16/2021 5:06:36 PM for display setting Display_Loc0x0_Res2560x1440__Display_Loc2560xM345_Res2560x1440
PersistentWindows: Restore finished in pass 4 with 15 windows recovered for display setting Display_Loc0x0_Res2560x1440__Display_Loc2560xM345_Res2560x1440
```
So I think you are right about it detecting a display resolution change. 
I did try pausing auto restore and the screens stay asleep when it is paused.
I do not think my method of waking the computer is the issue here, as I can leave the computer untouched as the screens go blank, and then get woken up again. If it is not PW doing this then I am not sure why it only happens while PW is running. I can provide a video if it helps.

**Comment by kangyu-california** [comment]
Thanks for the event log for diagnosis. The first 3 events indicate the monitor went through "off-on-off" status, by the time the last "off" event is received, the PW process to restore window layout to "on" status has already started, that explains this issue.

The next release will provide a solution for this. In the meantime, please keep running PW in "pause auto restore" status, and use the snapshot feature to restore window layout to previous status.

**Comment by kangyu-california** [comment]
Fix is implemented in release 5.20 by delay the response to monitor off-on-off events to 3 seconds.
If the default delay value still does not work, please specify a larger value through command line  -halt_restore  <seconds>

**Comment by evelyndooley** [comment]
So far this has been working as expected, thanks!
